torch/distributed/_tensor/api.py

```diff
@@ -270,15 +270,23 @@ def __tensor_flatten__(self):
         protocol to inform how to flatten a DTensor to local tensor
         for PT2 tracing
         """
-        return ["_local_tensor"], (self._spec, self.requires_grad)
+        return ["_local_tensor"], {
+            "_spec": self._spec,
+            "requires_grad": self.requires_grad,
+            # Any DTensor methods marked with @property need to go here,
+            # so dynamo can pick them upas proxy-able attributes
+            "placements": self._spec.placements,
+            "device_mesh": self._spec.mesh,
```

Comment:
Looks like we only change DTensor to use dict instead of tuple, I wonder this would break any other tensor subclass `__tensor_flatten__` or not, or dict/tuple is flexible for subclass authors?

Reply:
Late reply but answered here: https://github.com/pytorch/pytorch/pull/118209/files#r1490050517. I agree, I'm gonna change this fix so it doesn't require DTensor changes.